tls/s2n_resume.c

```diff
@@ -102,6 +104,73 @@ static int s2n_deserialize_resumption_state(struct s2n_connection *conn, struct
     return 0;
 }
 
+static int s2n_client_serialize_resumption_state(struct s2n_connection *conn, struct s2n_stuffer *to)
+{
+    /* Serialize session id */
+    GUARD(s2n_stuffer_write_uint8(to, S2N_STATE_WITH_SESSION_ID));
+    GUARD(s2n_stuffer_write_uint8(to, conn->session_id_len));
+    GUARD(s2n_stuffer_write_bytes(to, conn->session_id, conn->session_id_len));
+
+    /* Serialize session state */
+    GUARD(s2n_serialize_resumption_state(conn, to));
+
+    return 0;
+}
+
+static int s2n_client_deserialize_session_state(struct s2n_connection *conn, struct s2n_stuffer *from)
+{
+    uint8_t session_id_len;
+    GUARD(s2n_stuffer_read_uint8(from, &session_id_len));
+
+    if (session_id_len > S2N_TLS_SESSION_ID_MAX_LEN || session_id_len > s2n_stuffer_data_available(from)) {
```

Comment:
We should probably reject 0 length as well.

Reply:
Correct, updated.